klog.go

```diff
@@ -944,16 +951,20 @@ func (l *loggingT) output(s severity, log *logr.Logger, buf *buffer, depth int,
 			// will point to the same file, so just use one of them to write data.
 			if l.file[infoLog] == nil {
 				if err := l.createFiles(infoLog); err != nil {
-					os.Stderr.Write(data) // Make sure the message appears somewhere.
+					os.Stderr.Write(data)        // Make sure the message appears somewhere.
+					logExitFunc = func(error) {} // Set `logExitFunc` non-nil to unlock safely.
 					l.exit(err)
+					return
```

Comment:
Why do you want to return here?

I don't understand why this part of the function gets changed.